src/yuzu/mini_dump.cpp

```diff
@@ -0,0 +1,195 @@
+#include <cstdio>
+#include <cstring>
+#include <ctime>
+#include <filesystem>
+#include <windows.h>
+#include "yuzu/mini_dump.h"
+#include "yuzu/startup_checks.h"
+
+// dbghelp.h must be included after windows.h
+#include <dbghelp.h>
+
+void CreateMiniDump(HANDLE process_handle, DWORD process_id, MINIDUMP_EXCEPTION_INFORMATION* info,
+                    EXCEPTION_POINTERS* pep) {
+    char file_name[255];
+    const std::time_t the_time = std::time(nullptr);
+    std::strftime(file_name, 255, "yuzu-crash-%Y%m%d%H%M%S.dmp", std::localtime(&the_time));
+
+    // Open the file
+    HANDLE file_handle = CreateFileA(file_name, GENERIC_READ | GENERIC_WRITE, 0, nullptr,
+                                     CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, nullptr);
+
+    if (file_handle == nullptr || file_handle == INVALID_HANDLE_VALUE) {
+        std::fprintf(stderr, "CreateFileA failed. Error: %d\n", GetLastError());
+        return;
+    }
+
+    // Create the minidump
+    const MINIDUMP_TYPE dump_type = MiniDumpNormal;
+
+    const bool write_dump_status = MiniDumpWriteDump(process_handle, process_id, file_handle,
+                                                     dump_type, (pep != 0) ? info : 0, 0, 0);
+
+    if (write_dump_status) {
+        std::fprintf(stderr, "MiniDump created: %s\n", file_name);
+    } else {
+        std::fprintf(stderr, "MiniDumpWriteDump failed. Error: %d\n", GetLastError());
+    }
+
+    // Close the file
+    CloseHandle(file_handle);
+}
+
+void DumpFromDebugEvent(DEBUG_EVENT& deb_ev, PROCESS_INFORMATION& pi) {
+    EXCEPTION_RECORD& record = deb_ev.u.Exception.ExceptionRecord;
+
+    HANDLE thread_handle = OpenThread(THREAD_GET_CONTEXT, false, deb_ev.dwThreadId);
+    if (thread_handle == nullptr) {
+        std::fprintf(stderr, "OpenThread failed (%d)\n", GetLastError());
+    }
+
+    // Get child process context
+    CONTEXT context;
```

Comment:
```suggestion
    CONTEXT context = {};
```

Will let you get rid of the memset below